features/steps/thens/alignment.py

```diff
@@ -352,3 +359,61 @@ def step_impl(context, inst, activation_phrase):
                         observed=observed_msg,
                         severity=OutcomeSeverity.ERROR,
                     )
+
+
+@gherkin_ifc.step('Each segment must have geometric continuity in {continuity_type}')
+def step_impl(context, inst, continuity_type):
+    """
+    Assess geometric continuity between alignment segments for ALS016, ALS017, and ALS018
+    """
+    # Ref: https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcTransitionCode.htm
+    position_transition_codes = ["CONTINUOUS", "CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+    tangency_transition_codes = ["CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+
+    length_unit_scale_factor = ifcopenshell.util.unit.calculate_unit_scale(
+        ifc_file=context.model,
+        unit_type="LENGTHUNIT"
+    )
+    for previous, current in inst:
+        entity_contexts = ifc.recurrently_get_entity_attr(context, current, 'IfcRepresentation', 'ContextOfItems')
+        precision = ifc.get_precision_from_contexts(entity_contexts)
+
+        # calculate number of significant figures to display
+        # use the precision of the geometric context plus one additional digit to accommodate rounding
+        from math import ceil, log10
+        display_sig_figs = abs(int(ceil(log10(precision)))) + 1
+
+        continuity_type = continuity_type.lower()
+
+        if (continuity_type == "position") and (current.Transition in position_transition_codes):
+            expected = precision
+            observed = geometry.alignment_segment_positional_difference(
+                length_unit_scale_factor,
+                previous,
+                current
+            )
+
+        elif (continuity_type == "tangency") and (current.Transition in tangency_transition_codes):
+            expected = math.atan2(precision, current.SegmentLength.wrappedValue)
+            observed = geometry.alignment_segment_angular_difference(
+                length_unit_scale_factor,
+                previous,
+                current
+            )
+        else:
+            return
+
+        if abs(observed) > expected:
+            yield ValidationOutcome(
+                inst=current,
+                expected={
+                    "expected": expected,
+                    "num_digits": display_sig_figs,
+                    "context": context,
+                },
+                observed={
+                    "observed": observed,
+                    "num_digits": display_sig_figs,
+                    "context": context,
```

Comment:
Do we want to pass the whole context over? 
In the backend, it's not needed as we have this available in the decorator. In the frontend, it's perhaps not so useful information 
![image](https://github.com/user-attachments/assets/dd3bc89e-a8d3-450c-a06c-43dd4819a196)

Reply:
Ah, sorry, I now understand the purpose after seeing Thomas' comment. I think there is a confusion between two types of contexts:
- The behave Context class that we're using in Gherkin to transfer, for example, stack attributes.
- The context for parsing in frontend. 

For the latter, please refer to the PRs from a little while ago regarding[ adding context in Gherkin](https://github.com/buildingSMART/ifc-gherkin-rules/pull/185/files#diff-eddbd16c15972064cf608a2e27ef9ccc873a806fa85084208c1d59e1028846f6) and [parsing on the frontend](https://github.com/buildingSMART/validate/pull/65/files). 
We'd probably need to do something similar here

Reply:
Done, see https://github.com/buildingSMART/validate/pull/99